cleanup_b4_pr_discussion.md

```diff
@@ -0,0 +1,43 @@
+# Cleaning up commit history before creating a pull request
+
+It is important to realize that the frequency of and reasons for creating a commit vary from developer to
+developer. On the high commit frequency end of that spectrum we find developers using
+TDD (Test Driven Development). It is often the practice with TDD to create a commit each time a new
+test is written and the code is completed that causes this new test (and all other tests) to pass. This is 
```

Comment:
> It is often the practice with TDD to create a commit each time a new test is written and the code is completed that causes this new test (and all other tests) to pass

Does this mean that there are two commits or just one?

Reply:
The frequency of making a commit is a personal choice. When I do TDD, I will always commit after adding a test and getting it to pass. Sometimes, when writing the test and getting it to pass make take a while (multiple edits) I will also commit when the new test is completed and before I start making changes to allow the test to pass. (Should I add some text to explain that commit frequency is a personal choice?)

Reply:
Alright, i was just thinking that it's possible to interpret the sentence in two ways:

> It is often the practice with TDD to create a commit each time {a new test is written and the code is completed that causes this new test (and all other tests) to pass}

or alternatively

> It is often the practice with TDD to create a commit each time {a new test is written} and [when] {the code is completed that causes this new test (and all other tests) to pass}

but now that i read it again i'm guessing that the first is correct since there is no "when" added, have i understood?

Reply:
I will reword that section to make it less confusing.